src/cmd/flux-mini.py

```diff
@@ -0,0 +1,335 @@
+            task_count_dict = {"per_slot": 1}
+        slot = create_slot("task", num_slots, [core])
+        resource_section = create_resource("node", num_nodes, [slot])
+    else:
+        task_count_dict = {"per_slot": 1}
+        slot = create_slot("task", num_tasks, [core])
+        resource_section = slot
+
+    jobspec = {
+        "version": 1,
+        "resources": [resource_section],
+        "tasks": [
+            {
+                "command": command,
+                "slot": "task",
+                "count": task_count_dict,
+                "attributes": {},
+            }
+        ],
+        "attributes": {"system": {"cwd": os.getcwd(), "environment": environ}},
+    }
+    if walltime:
+        jobspec["attributes"]["system"]["duration"] = walltime
+
+    return jobspec
+
+
+def common_validate_args(args):
+    if args.nodes > args.ntasks:
+        raise ValueError("Number of nodes greater than the number of tasks")
+
+    if (
+        args.time
+        and re.match(r"^(\d+-)?\d+:\d+:\d+$", args.time) is None
+        and re.match(r"^\d+(:\d+)?$", args.time) is None
+    ):
+        raise ValueError(
+            "invalid time limit string format. "
+            "Acceptable formats include minutes[:seconds], [days-]hours:minutes:seconds"
+        )
+
+
+#  Convert walltime string to floating point seconds
+def walltime_to_duration(time):
+    if not time:
+        return None
+    p1 = re.compile(
+        r"^((?P<days>\d+)-)?"
+        r"(?P<hours>\d+):"
+        r"(?P<minutes>\d+):"
+        r"(?P<seconds>\d+)$"
+    )
+    p2 = re.compile(r"^(?P<minutes>\d+)(:(?P<seconds>\d+))?$")
+
+    t = 0.0
+    m = p2.search(time) or p1.search(time)
+    if m is None:
+        return None
+    vals = {k: float(v) for k, v in m.groupdict().items() if v is not None}
+    if "days" in vals:
+        t = t + vals["days"] * 60 * 60 * 24
+    if "hours" in vals:
+        t = t + vals["hours"] * 60 * 60
+    if "minutes" in vals:
+        t = t + vals["minutes"] * 60
+    if "seconds" in vals:
+        t = t + vals["seconds"]
+    if t == 0.0:
+        return None
+    return t
+
+
+def kv_list_split(s):
+    """
+    Split a key/value list with optional values, e.g. 'FOO,BAR=baz,...'
+    and return a dict.
+    """
+    return dict((a, b) for a, _, b in [x.partition("=") for x in s.split(",")])
+
+
+def common_get_parser():
+    """
+    Arg parser for subcommand - common options for run and submit
+    """
+    parser = argparse.ArgumentParser(add_help=False)
+    parser.add_argument(
+        "-N", "--nodes", type=int, default=0, help="Number of nodes to allocate"
+    )
+    parser.add_argument("-n", "--ntasks", type=int, help="Number of tasks to start")
+    parser.add_argument(
+        "-c",
+        "--cpus-per-task",
+        type=int,
+        default=1,
+        help="Number of cpus to allocate per task",
+    )
+    parser.add_argument(
+        "-t", "--time", help="Time limit: min[:sec] or [days-]hours:min:sec"
```

Comment:
Since `mini run` doesn't need to abide by Slurm's time format, do we want to be consistent with `flux run` and use the Flux Standard Duration?  [(Discussion on time in `flux run` PR)](https://github.com/flux-framework/flux-core/issues/2213)

Reply:
Good idea!  Just pushed a fixup.

I grabbed @trws's `parse_fsd()` from #2323 and changed it to raise a ValueError if input is not valid.  This is a good candidate function for adding to the python bindings IMHO.

Reply:
One of the two would definitely be good.  The bindings may actually have access to the parse_fsd function in libutil already, but this way we avoid the string conversion etc.  Either way, but we should have it in flux.util or some such for sure.

Reply:
Just added a new issue to track the `parse_fsd` function in python: https://github.com/flux-framework/flux-core/issues/2393